### PR TITLE
[CAMEL-18527] Camel Kafka Component: batch producer with individual headers

### PR DESCRIPTION
**Pull request by rwiedmann**
I hope this PR looks better than the previous one ;-)

### CODE REVIEW

**Comment by oscerd** [comment]
Usually we open PR against main branch and then is up to maintainer to eventually backport. It shouldn't be a problem for this, maybe for the next time.

File: components/camel-kafka/src/main/java/org/apache/camel/component/kafka/producer/support/PropagatedHeadersProvider.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import org.apache.camel.Exchange;
+import org.apache.camel.Message;
+import org.apache.camel.component.kafka.KafkaConfiguration;
+import org.apache.camel.component.kafka.KafkaProducer;
+import org.apache.kafka.common.header.Header;
+
+/**
+ * Used to provide individual kafka header values if feature "batchWithIndividualHeaders" is enabled.
+ *
+ * @author r.wiedmann@bitproquo.org
```
Comment (davsclaus) [inline components/camel-kafka/src/main/java/org/apache/camel/component/kafka/producer/support/PropagatedHeadersProvider.java:30]: Remove author tag

**Review by davsclaus — CHANGES_REQUESTED** [review]

File: components/camel-kafka/src/main/java/org/apache/camel/component/kafka/producer/support/PropagatedHeadersProvider.java (hunk 1 above)
Comment (rwiedmann) [reply components/camel-kafka/src/main/java/org/apache/camel/component/kafka/producer/support/PropagatedHeadersProvider.java]: done

**Review by oscerd — APPROVED** [review]

**Review by davsclaus — APPROVED** [review]

**Comment by davsclaus** [comment]
merged to main also

**Comment by rwiedmann** [comment]
Thank you,

I'm very impressed with how quickly the team responded!

**Pull request merged**